lib/src/async_import_cache.dart

```diff
@@ -181,17 +168,34 @@ final class AsyncImportCache {
     // `canonicalize()` calls we've attempted are cacheable. Only if they are do
     // we store the result in the cache.
     var cacheable = true;
-    for (var importer in _importers) {
+    for (var i = 0; i < _importers.length; i++) {
+      var importer = _importers[i];
```

Comment:
There should be a check around here for the presence of `_perImporterCanonicalizeCache`, no? 

Otherwise `_canonicalize` gets executed for all the importers previous to the one that returned a non-null and non-cacheable result

Reply:
Yeah, actually checking the cache is an important part of a caching scheme :sweat_smile: